x-pack/plugins/rule_registry/server/alert_data_client/alerts_client.ts

```diff
@@ -0,0 +1,192 @@
+import { ParsedTechnicalFields } from '../../common/parse_technical_fields';
+
+export interface ConstructorOptions {
+  logger: Logger;
+  authorization: PublicMethodsOf<AlertingAuthorization>;
+  auditLogger?: AuditLogger;
+  esClient: ElasticsearchClient;
+  ruleDataService: PublicMethodsOf<RuleDataPluginService>;
+}
+
+export interface UpdateOptions<Params extends AlertTypeParams> {
+  id: string;
+  data: {
+    status: string;
+  };
+  // observability-apm see here: x-pack/plugins/apm/server/plugin.ts:191
+  indexName: string;
+}
+
+interface GetAlertParams {
+  id: string;
+  // observability-apm see here: x-pack/plugins/apm/server/plugin.ts:191
+  indexName: string;
+}
+
+export class AlertsClient {
+  private readonly logger: Logger;
+  private readonly auditLogger?: AuditLogger;
+  private readonly authorization: PublicMethodsOf<AlertingAuthorization>;
+  private readonly esClient: ElasticsearchClient;
+  private readonly ruleDataService: PublicMethodsOf<RuleDataPluginService>;
+
+  constructor({
+    auditLogger,
+    authorization,
+    logger,
+    esClient,
+    ruleDataService,
+  }: ConstructorOptions) {
+    this.logger = logger;
+    this.authorization = authorization;
+    this.esClient = esClient;
+    this.auditLogger = auditLogger;
+    this.ruleDataService = ruleDataService;
+  }
+
+  public getFullAssetName() {
+    return this.ruleDataService?.getFullAssetName();
+  }
+
+  public async getAlertsIndex(featureIds: string[]) {
+    return this.authorization.getAuthorizedAlertsIndices(
+      featureIds.length !== 0 ? featureIds : ['apm', 'siem']
+    );
+  }
+
+  private async fetchAlert({ id, indexName }: GetAlertParams): Promise<ParsedTechnicalFields> {
+    try {
+      const result = await this.esClient.get<ParsedTechnicalFields>({
+        index: indexName,
+        id,
+      });
+
+      if (
+        result.body._source == null ||
+        result.body._source[RULE_ID] == null ||
+        result.body._source[OWNER] == null
+      ) {
+        const errorMessage = `[rac] - Unable to retrieve alert details for alert with id of "${id}".`;
+        this.logger.debug(errorMessage);
+        throw new Error(errorMessage);
+      }
+
+      return result.body._source;
+    } catch (error) {
+      const errorMessage = `[rac] - Unable to retrieve alert with id of "${id}".`;
+      this.logger.debug(errorMessage);
+      throw error;
+    }
+  }
+
+  public async get({ id, indexName }: GetAlertParams): Promise<ParsedTechnicalFields> {
+    try {
+      // first search for the alert by id, then use the alert info to check if user has access to it
+      const alert = await this.fetchAlert({
+        id,
+        indexName,
+      });
+
+      // this.authorization leverages the alerting plugin's authorization
+      // client exposed to us for reuse
+      await this.authorization.ensureAuthorized({
+        ruleTypeId: alert['rule.id']!, // we assert in fetchAlert that these values are non-null
```

Comment:
Can we use the [constants](https://github.com/elastic/kibana/blob/bdaea87aa65ea0a5efe588d6bd643e1e17c9ed4b/packages/kbn-rule-data-utils/src/technical_field_names.ts#L17-L18) here?